mininterface/tk_interface/utils.py

```diff
@@ -67,12 +68,89 @@ def get(self):
 
 
 def choose_file_handler(variable: Variable, tag: PathTag):
+    """Handler for file/directory selection on PathTag"""
     def _(*_):
-        if tag.multiple:
-            out = str(list(askopenfilenames(title="Choose files")))
+        initialdir = os.getcwd()
+
+        # Check whether this is a directory selection
+        if tag.is_dir:
+            # Directory selection using askdirectory
+            kwargs = {"title": "Select Directory", "initialdir": initialdir}
+            selected_dir = askdirectory(**kwargs)
+            if not selected_dir:  # User cancelled
+                return
+
+            if tag.multiple:
+                # Handle multiple selection for directories
+                try:
+                    current_val = variable.get()
```

Comment:
Instead of this whole and complicated stuff (which must have been a lot of work for you), just lean on the tag.val for reading. For tag.multiple, there will be a list of str|Path. For not tag.multiple, there is str|Path.

Also, I think you don't need to cast it to str, just you variable.set([Path(...), Path(...)]) and it will work, it will later convert to tag.val automatically. We already have some generic mechanisms to convert the values through tkinter.variable back and forth.